### ISSUE TITLE
Missing version 1.18

### ISSUE THREAD

**Issue by dhouck** [ImUrX/libenchcrack#1]
There is no option for cracking enchantments in 1.18.

I donʼt know if this is an issue for this (if 1.18 is meaningfully different) or [enchcracker](https://github.com/ImUrX/enchcracker) (if it isnʼt and the `v1.16 - v1.17.1` text just needs updating).  Iʼm guessing itʼs for here because itʼs failing to crack my seed, but that could be user error.

**Comment by ImUrX** [comment]
From what I know it shouldn't fail but let me verify that first

**Comment by ImUrX** [comment]
And just checked, there been no changes on how RNG works. The versions are for manipulating enchantments not for cracking :p.

I'm thinking of adding some kind of checklist to help people crack and dont lose themselves in the middle of the bookshelf hell but thats more about enchcracker than the lib

**Comment by dhouck** [comment]
Okay, cool, makes sense.  And I think I know what I did wrong; in one case it was being mistaken about how many bookshelves were blocked, and in another it was the `/give` command apparently being one of the things that messes with the seed.

I donʼt know how hard this is to update for different versions, but I expect it breaks in the latest snapshot because thereʼs a new enchantment?  If thatʼs the case and I can figure out how this works (and you donʼt get to it first) would you mind a PR for adding the snapshot, or do you want to stick to full versions only?  Iʼm not saying I will, but I am trying to learn Rust and this seems like a good project to try with.

**Comment by ImUrX** [comment]
Yeah sure, go for it

You will need to add a new version for the enchantment and etc, the api is pretty understandable imo

**Comment by dhouck** [comment]
Iʼm no longer convinced youʼre correct that this didnʼt change.  I tested it in 1.17.1 and it worked, but now it doesnʼt.  Iʼm in 1.18.2, and my XP seed is (according to NBT Explorer) 1986994563 (hex 766F2183), and the website is failing to crack my seed.

In particular, Iʼm getting that (for a wooden sword) my enchantment levels are
| Bookshelves | 1st | 2nd | 3rd |
| ----------: | --- | --- | --- |
|          15 |  9  |  11 |  30 |
|          14 |  8  |  15 |  28 |
|          13 |  5  |  13 |  26 |
|          12 |  4  |  10 |  24 |
|          11 |  4  |  9  |  22 |
|          10 |  4  |  11 |  20 |
|           9 |  4  |  6  |  18 |
|           8 |  5  |  9  |  16 |
|           7 |  5  |  6  |  14 |
|           6 |  4  |  6  |  13 |
|           5 |  3  |  3  |  10 |
|           4 |  3  |  5  |  8  |
|           3 |  3  |  3  |  6  |
|           2 |  2  |  3  |  7  |
|           1 |  2  |  2  |  5  |
|           0 |  2  |  -  |  5  |

When I enter all the information form rows 15 through 10 into https://imurx.github.io/enchcracker/, it says “No possible seeds”.  This is consistent throughout three attempts where I looked at the numbers each time; itʼs not a transcription error.  Iʼm not enchanting anything (or doing any of the other forbidden activities, not that it should matter until later when it assumes my XP seed is consecutive with a different one).  Iʼm blocking bookshelves by using a contraption I built beforehand to lower them with sticky pistons; thereʼs no chance Iʼm mis-counting how many are removed (and besides, each row of this table is unique, so I canʼt have made a mistake there).

Here are screenshots showing the enchanting table for each of levels 15 through 10:
![2022-03-26_04 12 37](https://user-images.githubusercontent.com/524925/160233120-1540101a-bf5a-49eb-8324-af6bb84e4cd3.png)
![2022-03-26_04 12 46](https://user-images.githubusercontent.com/524925/160233122-de41acc7-3ce9-4999-ab95-c60020af90eb.png)
![2022-03-26_04 12 54](https://user-images.githubusercontent.com/524925/160233123-e7e35453-6ffe-497b-b9f5-f5548380f54f.png)
![2022-03-26_04 13 03](https://user-images.githubusercontent.com/524925/160233124-b297efe9-1c16-470c-8d42-473f8ad59b7a.png)
![2022-03-26_04 13 10](https://user-images.githubusercontent.com/524925/160233126-de519eb0-bf68-4a01-af8d-de0771ae2632.png)
![2022-03-26_04 13 25](https://user-images.githubusercontent.com/524925/160233127-a8a9c032-ae18-44a5-993a-06d5e4f546ea.png)

**Comment by ImUrX** [comment]
Sorry if it's too much but can you record a video of trying to crack the seed? Minecraft didn't change the RNG in 1.18.2 so it's illogical it shouldn't work. You can even try it doing it with Earthcomputer's EnchantmentCracker

**Comment by dhouck** [comment]
Sure, I can try in a few hours, but the one previous time Iʼve tried to use OBS Studio, it misbehaved in a weird way, and AFAIK Replay Mod doesnʼt record player GUIs still.  If nothingʼs changed with my OBS situation, the video will have poor or no audio, but should otherwise be fine.

Also, I *think* I gave enough information to reproduce it yourself, if I have more trouble than expected making the video?

**Comment by ImUrX** [comment]
When I closed the issue I tested enchcracker in 1.18.2 myself and I cracked the seed and manipulated it successfully.

Now I'm trying again and it went... succesfully :/

**Comment by ImUrX** [comment]
There is no problem with the audio, I just need to see if you dont actually change the RNG in any part

**Comment by dhouck** [comment]
Okay, still canʼt do the video right now because Iʼm doing other things but Iʼll get to it in a couple hours.  Iʼm also removing all my mods like MiniHud, even though none of them should affect gameplay at all and certainly not enchantments, because itʼs good to eliminate variables.

**Comment by dhouck** [comment]
I have made the video (below, although it isnʼt relevant), but while writing out the description of the video I realized a potential issue: the `Reset` button on the website either doesnʼt work or doesnʼt do what I think it does; when I do the website in incognito mode (using the numbers I got in the video) it works (at least as far as correctly cracking the xp seeds).

The website isnʼt shown in the video, only my Minecraft window, so I donʼt know how useful this will be, but [hereʼs the video anyway](https://youtu.be/2lreoSH6Z8k).  For some reason beyond my understanding, the audio worked perfectly this time.
